util/postprocess-makedepend.pl

```diff
@@ -0,0 +1,134 @@
+
+            return $line;
+        },
+    'VMS C' =>
+        sub {
+            my $line = shift;
+
+            # current versions of DEC / Compaq / HP / VSI C strips away all
+            # directory information from the object file, so we must insert it
+            # back. Just to be safe against future changes, we check that there
+            # really is no directory information.
+            my $directory = shift;
+
+            # The pattern for target and dependencies will always take this
+            # form:
+            #
+            #   target SPACE : SPACE deps
+            #
+            # This is so a volume delimiter (a : without any spaces around it)
+            # won't get mixed up with the target / deps delimiter.  We use this
+            # fact in the regexp below to make sure we do look at the target.
+            $line =~ s/^/$directory/ unless /^\S+[:>\]]\S+\s+:/;
+
+            # We know that VMS has system header files in text libraries,
+            # extension .TLB.  We also know that our header files aren't stored
+            # in text libraries.  Finally, we know that VMS C produces exactly
+            # one dependency per line, so we simply discard any line ending with
+            # .TLB.
+            return undef if /\.TLB\s*$/;
+
+            return $line;
+        },
+    'VC' =>
+        sub {
+            my $line = shift;
+            my $object = shift;
+
+            # For the moment, we only support Visual C on native Windows, or
+            # compatible compilers.  With those, the flags /Zs /showIncludes
+            # give us the necessary output to be able to create dependencies
+            # that nmake (or any 'make' implementation) should be able to read,
+            # with a bit of help.  The output we're interested in looks like
+            # this (it always starts the same)
+            #
+            #   Note: including file: {whatever header file}
+            #
+            # So all we really have to do is to is to replace the start of the
+            # line with an object file specification, given to us as an extra
+            # argument (passed from $ARGV[1]);
+            #
+            # There are also other lines mixed in, for example compiler
+            # warnings, so we simply discard anything that doesn't start with
+            # the Note:
+
+            if (/^Note: including file: */) {
+                (my $tail = $') =~ s/\s*\R$//;
+
+                # VC gives us absolute paths for all include files, so to
+                # remove system header dependencies, we need to check that
+                # they don't match $abs_srcdir or $abs_blddir
+                $tail = canonpath($tail);
+                my $substr1 = substr($tail, 0, length $abs_srcdir);
+                my $substr2 = substr($tail, 0, length $abs_blddir);
+
+                if (lc $abs_srcdir eq lc $substr1
+                        || lc $abs_blddir eq lc $substr2) {
```

Comment:
Wouldn't `$tail !~ m|^$abs_scrdir|i && $tail !~ m|^$abs_blddir|i` do?

Reply:
Can you guarantee that they contain safe characters?  I can't.

Reply:
`\Q` `\E`

Reply:
> \Q \E

no, of course not...

Reply:
> \Q \E

Oh yes, \Q \E works, but original condition should be reversed, i.e. `$tail \~ m|^\Q$abs_scrdir\E|i || $tail =~ m|^\Q$abs_blddir\E|i`.

Reply:
Actually, that should work, shouldn't it?  Within what the perl version in question supports re non-ASCII characters and all that...  but still, `\Q` and `\E` existed before 5.10, and were functional as far as I know.  Do you have a different experience?

Reply:
Ah, now I see that you figured that out...

Reply:
Fixed